src/ol/renderer/maprenderer.js

```diff
@@ -171,6 +171,26 @@ ol.renderer.Map.prototype.forEachFeatureAtPixel =
 
 
 /**
+ * @param {ol.Coordinate} coordinate Coordinate.
+ * @param {olx.FrameState} frameState FrameState.
+ * @param {function(this: U, ol.layer.Layer): boolean} layerFilter Layer filter
+ *     function, only layers which are visible and for which this function
+ *     returns `true` will be tested for features.  By default, all visible
+ *     layers will be tested.
+ * @param {U} thisArg Value to use as `this` when executing `layerFilter`.
+ * @return {boolean} Is there a feature at the given pixel?
+ * @template U
+ */
+ol.renderer.Map.prototype.hasFeatureAtPixel =
+    function(coordinate, frameState, layerFilter, thisArg) {
+  var hasFeature = this.forEachFeatureAtPixel(
+      coordinate, frameState, goog.functions.TRUE, this, layerFilter, thisArg);
```

Comment:
I was expecting to see something like this at the map renderer level:
```js
ol.renderer.Map.prototype.hasFeatureAtPixel =
    function(coordinate, frameState, layerFilter, thisArg) {
  var hasFeature = false;
  var layerStates = this.getMap().getLayerGroup().getLayerStatesArray();
  for (var i = layerStates.length - 1; i >= 0; --i) {
    var layerState = layerStates[i];
    var layer = layerState.layer;
    if (ol.layer.Layer.visibleAtResolution(layerState, viewState.resolution) &&
        layerFilter.call(thisArg2, layer)) {
      var layerRenderer = this.getLayerRenderer(layer);
      hasFeature = layerRenderer.hasFeatureAtPixel(coordinate, frameState);
      if (hasFeature) {
        return true;
      }
    }
  }
  return hasFeature;
};
```

And then see the default layer renderer implementation of `hasFeatureAtPixel` delegate to `forEachFeatureAtPixel`.

But I see now that this doesn't account for feature overlays and the specific work that the WebGL renderer has to do there.  This can happen later, but I still can imagine that we could have one `hasFeatureAtPixel` implementation at the map renderer level that could call a specific `overlayHasFeatureAtPixel` method for each map renderer type.